caikit_nlp/toolkit/text_generation/tgis_utils.py

```diff
@@ -683,3 +688,33 @@ def unary_tokenize(
         return TokenizationResults(
             token_count=response.token_count,
         )
+
+
+def get_route_info(
+    context: Optional[RuntimeServerContextType],
+) -> Optional[str]:
+    """
+    Returns a tuple `(True, x-route-info)` from context if "x-route-info" was found in
+    the headers/metadata.
+
+    Otherwise returns a tuple `(False, None)` if "x-route-info" was not found in the
+    context or if context is None.
+    """
+    if context is None:
+        return None
+
+    if isinstance(context, grpc.ServicerContext):
+        route_info = dict(context.invocation_metadata()).get(ROUTE_INFO_HEADER_KEY)
+        if route_info:
+            return route_info
+    elif isinstance(context, fastapi.Request):
+        route_info = context.headers.get(ROUTE_INFO_HEADER_KEY)
+        if route_info:
+            return route_info
+    else:
+        error.log_raise(
+            "<NLP92615097E>",
+            ValueError(f"context is of an unsupported type: {type(context)}"),
+        )
+
+    return None
```

Comment:
NIT: this is both implicit and unreachable since `else` will raise. I'm guessing it's here because your IDE doesn't know that?

Reply:
Done. Removed.